<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Generating Dynamic DEA Reports with the prolfqua R Package • prolfquapp</title>
<script src="deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="deps/headroom-0.11.0/headroom.min.js"></script><script src="deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="deps/search-1.0.0/fuse.min.js"></script><script src="deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="pkgdown.js"></script><meta property="og:title" content="Generating Dynamic DEA Reports with the prolfqua R Package">
<meta name="description" content="The prolfquapp is a user-friendly application to streamline protein differential expression analysis (DEA) in core facilities. The application leverages the preprocessing methods and statistical models implemented in the R package prolfqua. It generates dynamic HTML reports that contain quality control plots and visualizations of the DEA results. The prolfquapp also exports results in multiple formats, including XLSX files, .rnk or txt files for gene set enrichment analysis, and Bioconductor SummarizedExperiment format for import into interactive visualization tools like OmicsViewer and iSEE. The prolfquapp offers a comprehensive and efficient solution for researchers seeking to analyze protein differential expression data in a core facility setting.">
<meta property="og:description" content="The prolfquapp is a user-friendly application to streamline protein differential expression analysis (DEA) in core facilities. The application leverages the preprocessing methods and statistical models implemented in the R package prolfqua. It generates dynamic HTML reports that contain quality control plots and visualizations of the DEA results. The prolfquapp also exports results in multiple formats, including XLSX files, .rnk or txt files for gene set enrichment analysis, and Bioconductor SummarizedExperiment format for import into interactive visualization tools like OmicsViewer and iSEE. The prolfquapp offers a comprehensive and efficient solution for researchers seeking to analyze protein differential expression data in a core facility setting.">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="index.html">prolfquapp</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">2.0.10</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="articles/prolfquapp.html">Get started</a></li>
<li class="nav-item"><a class="nav-link" href="reference/index.html">Reference</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="articles/Auxiliary_ExDesignSurvey.html">Auxiliary - Bioinformatics Project Meeting Agenda</a></li>
    <li><a class="dropdown-item" href="articles/QCandSampleSize.html">Quality Control &amp; Sample Size Estimation</a></li>
  </ul>
</li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/prolfqua/prolfquapp/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-home">
<div class="row">
  <main id="main" class="col-md-9">
<div class="section level1">
<div class="page-header"><h1 id="prolfquapp-ˌproʊˈlɛfkəˌæp-generating-dynamic-dea-reports-using-a-command-line-interface-to-the-prolfqua-r-package">prolfquapp (ˌproʊˈlɛf.kə.ˌæp): Generating Dynamic DEA Reports using a command line interface to the prolfqua R Package<a class="anchor" aria-label="anchor" href="#prolfquapp-%CB%8Cpro%CA%8A%CB%88l%C9%9Bfk%C9%99%CB%8C%C3%A6p-generating-dynamic-dea-reports-using-a-command-line-interface-to-the-prolfqua-r-package"></a>
</h1></div>
<p>Read on JPR <a href="https://pubs.acs.org/doi/10.1021/acs.jproteome.4c00911" class="external-link uri">https://pubs.acs.org/doi/10.1021/acs.jproteome.4c00911</a> “prolfquapp ─ A User-Friendly Command-Line Tool Simplifying Differential Expression Analysis in Quantitative Proteomics”</p>
<p>Welcome to <em>prolfquapp</em> on GitHub! Here, you’ll find everything you need to elevate your protein differential expression analysis. <em>Prolfquapp</em> integrates powerful preprocessing methods and advanced statistical models from the <em>prolfqua</em> R package <a href="https://github.com/fgcz/prolfqua" class="external-link">prolfqua</a> <a href="https://pubs.acs.org/doi/10.1021/acs.jproteome.2c00441" class="external-link">doi.org/10.1021/acs.jproteome.2c00441</a> to deliver insightful, clear visualizations and robust data outputs. Generate dynamic HTML reports, versatile file formats, and dive into interactive data visualization with <a href="https://github.com/fgcz/exploreDE" class="external-link">ExploreDE</a>. <em>Prolfquapp</em> implements a command line interface to run protein differential expression analysis, that can be integrated into your workflow manager.</p>
<div class="float">
<img src="https://github.com/prolfqua/prolfquapp/blob/master/inst/poster/Prolfqapp_Highlight.png?raw=true" alt="prolfquapp"><div class="figcaption">prolfquapp</div>
</div>
</div>
<div class="section level1">
<h1 id="differential-expression-analysis-workflow-with-prolfquapp">Differential Expression Analysis Workflow with prolfquapp<a class="anchor" aria-label="anchor" href="#differential-expression-analysis-workflow-with-prolfquapp"></a>
</h1>
<p>After running your Quantification software, DIA-NN, MAXQUANT, FragPipe-TMT, FragPipe-DIA or FragPipe-LFQ, the quantification results are in an <code>data_dir</code>. Please add the <code>.fasta</code> file which was used by the quantification software to the <code>data_dir</code>.</p>
<p><em>prolfquapp</em> is a set of command line tools.</p>
<p>To use it open your shell (linux, mac, or the linux subsystem on windows). Change into the directory with the identification/quantification results coming from FragPipe, MaxQuant, DIA-NN, Spectronaut etc. In the directory (in the terminal Linux or MAC) execute:</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb1-1"><a href="#cb1-1" tabindex="-1"></a><span class="ex">R</span> <span class="at">--vanilla</span> <span class="at">-e</span> <span class="st">"prolfquapp::copy_shell_script(workdir = '.')"</span> <span class="co"># if you have R on you system</span></span></code></pre></div>
<p>or, if you have <a href="https://www.docker.com/products/docker-desktop/" class="external-link">Docker</a> on you system, you can also download <a href="https://raw.githubusercontent.com/prolfqua/prolfquapp/refs/heads/master/inst/application/bin/prolfquapp_docker.sh" class="external-link">prolfquapp_docker.sh</a> and execute:</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb2-1"><a href="#cb2-1" tabindex="-1"></a><span class="ex">prolfquapp_docker.sh</span> R <span class="at">--vanilla</span> <span class="at">-e</span> <span class="st">"prolfquapp::copy_shell_script(workdir = '.')"</span> <span class="co"># if you wan to use the prolfqua docker image and container</span></span></code></pre></div>
<p>This will place the following four shell script files (linux), or bat files (windows) into your working directory:</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb3-1"><a href="#cb3-1" tabindex="-1"></a><span class="ex">[1]</span> <span class="st">"/&lt;working_directory&gt;/prolfqua_dea.sh"</span></span>
<span id="cb3-2"><a href="#cb3-2" tabindex="-1"></a><span class="ex">[3]</span> <span class="st">"/&lt;working_directory&gt;/prolfqua_yaml.sh"</span></span>
<span id="cb3-3"><a href="#cb3-3" tabindex="-1"></a><span class="ex">[4]</span> <span class="st">"/&lt;working_directory&gt;/prolfqua_qc.sh"</span></span>
<span id="cb3-4"><a href="#cb3-4" tabindex="-1"></a><span class="ex">[5]</span> <span class="st">"/&lt;working_directory&gt;/prolfqua_dataset.sh"</span></span></code></pre></div>
<p>On Linux give the executables LINUX permissions:</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb4-1"><a href="#cb4-1" tabindex="-1"></a><span class="fu">chmod</span> a+x prolfqua_<span class="pp">*</span></span></code></pre></div>
<p>All scripts can be run with the option <code>--help</code>.</p>
<div class="section level2">
<h2 id="workflow-overview">Workflow Overview<a class="anchor" aria-label="anchor" href="#workflow-overview"></a>
</h2>
<ol style="list-style-type: decimal">
<li><a href="#id_1-create-dataset">Create Dataset</a></li>
<li><a href="#id_2-generate-quality-control-qc">Generate Quality Control (QC)</a></li>
<li><a href="#id_3-generate-prolfqua-yaml">Generate prolfqua YAML</a></li>
<li><a href="#id_4-run-differential-expression-analysis">Run Differential Expression Analysis</a></li>
</ol>
</div>
<div class="section level2">
<h2 id="id_1-create-dataset">1. Create Dataset<a class="anchor" aria-label="anchor" href="#id_1-create-dataset"></a>
</h2>
<p>The first step involves preparing the dataset by providing the experiment annotation. This is done using the <code>prolfqua_dataset.sh</code> script.</p>
<ul>
<li>
<strong>Input</strong>: directory containing identification/quantification software outputs</li>
<li>
<strong>Output</strong>: csv, tsv or xlsx file template,</li>
</ul>
<p>To create a <code>prolfquapp</code> compatible experiment annotation file run:</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb5-1"><a href="#cb5-1" tabindex="-1"></a><span class="ex">./prolfqua_dataset.sh</span> <span class="at">-i</span> data_dir/ <span class="at">-s</span> DIANN <span class="at">-d</span> annotation.xlsx <span class="co"># if R and prolfuqapp is installed on your computer</span></span></code></pre></div>
<p>or if you use our <em>prolfquapp</em> docker container</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb6-1"><a href="#cb6-1" tabindex="-1"></a><span class="ex">./prolfquapp_docker.sh</span> prolfqua_dataset.sh <span class="at">-i</span> data_dir/ <span class="at">-s</span> DIANN <span class="at">-d</span> annotation.xlsx <span class="co"># if R you use the prolfquapp docker container</span></span></code></pre></div>
<p>The <code>annotation.xlsx</code> file will be generated, and will contain 5 columns.</p>
<ul>
<li><p>Relative.Path/Path/raw.file/channel/ (unique*)</p></li>
<li><p>name - used in tables and figures (unique*)</p></li>
<li><p>group/experiment/ - main factor</p></li>
<li><p>subject/bioreplicate (optional** or keep cells empty) - blocking factor</p></li>
<li><p>control - used to specify the control condition (C) (optional)</p></li>
<li><p>The rows must contain a unique value (no duplicates per column) ** If the experiment is not paired, or has no blocking factor (e.g. batch, cell line) delete the subject column.</p></li>
</ul>
<p>The column raw.file is already filled out, based on the information available in the input directory. You will need to fill out the missing columns, e.g. group, subject, control. The column names are not case sensitive.</p>
<p>The <code>annotation.xlsx</code> file will be generated, and you will need to fill out the empty cells.</p>
</div>
<div class="section level2">
<h2 id="id_2-generate-quality-control-qc">2. Generate Quality Control (QC)<a class="anchor" aria-label="anchor" href="#id_2-generate-quality-control-qc"></a>
</h2>
<p>The <code>prolfqua_qc.sh</code> script will create a QC report. The report consists of two HTML documents and XLSX file.</p>
<ul>
<li>
<strong>Input</strong>: Dataset from step 1 and directory containing identification/quantification software outputs</li>
<li>
<strong>Output</strong>: QC report and visualizations</li>
</ul>
<div class="sourceCode" id="cb7"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb7-1"><a href="#cb7-1" tabindex="-1"></a><span class="ex">./prolfqua_qc.sh</span> <span class="at">-i</span> data_dir/ <span class="at">-p</span> ProjectName <span class="at">-O</span> ordername <span class="at">-w</span> WorkunitName <span class="at">-d</span> annotation.xlsx <span class="at">-s</span> DIANN <span class="at">-o</span> where_to_write_results</span>
<span id="cb7-2"><a href="#cb7-2" tabindex="-1"></a><span class="co"># or prefix with ./prolfquapp_docker.sh to run the docker container</span></span>
<span id="cb7-3"><a href="#cb7-3" tabindex="-1"></a><span class="co"># ./prolfquapp_docker.sh prolfqua_qc.sh -i data_dir/ -p ProjectName -O ordername -w WorkunitName -d annotation.xlsx -s DIANN -o where_to_write_results</span></span></code></pre></div>
<p>This will generate a subfolder which starts with “QC_” with all the analysis results.</p>
</div>
<div class="section level2">
<h2 id="id_3-generate-prolfquapp-yaml">3. Generate prolfquapp yaml<a class="anchor" aria-label="anchor" href="#id_3-generate-prolfquapp-yaml"></a>
</h2>
<p>Using the <code>./prolfqua_yaml.R</code> command line tool you can set the parameters of the DEA and create a configuration file in YAML format.</p>
<ul>
<li>
<strong>Output</strong>: Yaml file</li>
</ul>
<div class="sourceCode" id="cb8"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb8-1"><a href="#cb8-1" tabindex="-1"></a><span class="ex">./prolfqua_yaml.sh</span> <span class="at">-y</span> config.yaml</span>
<span id="cb8-2"><a href="#cb8-2" tabindex="-1"></a><span class="co"># or prefix with ./prolfquapp_docker.sh to run the docker container</span></span>
<span id="cb8-3"><a href="#cb8-3" tabindex="-1"></a><span class="co"># ./prolfquapp_docker.sh prolfqua_yaml.sh -y config.yaml</span></span></code></pre></div>
<p>To see which parameters can be set using <code>prolfqua_yaml.sh</code> use the <code>-h</code> switch. Other parameters you can set by editing the yaml file.</p>
</div>
<div class="section level2">
<h2 id="id_4-run-differential-expression-analysis">4. Run Differential Expression Analysis<a class="anchor" aria-label="anchor" href="#id_4-run-differential-expression-analysis"></a>
</h2>
<p>Finally, the <code>prolfqua_dea.sh</code> script runs the differential expression analysis using the configuration file generated in the previous step.</p>
<ul>
<li>
<strong>Input</strong>: directory containing identification/quantification software outputs, dataset from step 1, configuration file from step 3.</li>
<li>
<strong>Output</strong>: folder containing differential expression analysis results (html files, excel tables, rank files, SummarizedExperiment.rds).</li>
</ul>
<p>After setting the parameters in the config.yaml file you can run the DEA analysis by:</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb9-1"><a href="#cb9-1" tabindex="-1"></a><span class="ex">./prolfqua_dea.sh</span> <span class="at">-i</span> data_dir/ <span class="at">-d</span> annotation.xlsx <span class="at">-y</span> config.yaml <span class="at">-w</span> NameOfAnalysis <span class="at">-s</span> DIANN</span>
<span id="cb9-2"><a href="#cb9-2" tabindex="-1"></a><span class="co"># and again you run the version within the docker container with</span></span>
<span id="cb9-3"><a href="#cb9-3" tabindex="-1"></a><span class="co"># ./prolfquapp_docker.sh prolfqua_dea.sh -i data_dir/ -d annotation.xlsx -y config.yaml -w NameOfAnalysis -s DIANN</span></span></code></pre></div>
<p>This will generate a subfolder which starts with “DEA_” and writes all the analysis results as well as the input data.</p>
</div>
<div class="section level2">
<h2 id="how-to-install">How to install<a class="anchor" aria-label="anchor" href="#how-to-install"></a>
</h2>
<p><strong>Linux</strong></p>
<div class="sourceCode" id="cb10"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb10-1"><a href="#cb10-1" tabindex="-1"></a><span class="bu">export</span> <span class="va">R_LIBS_SITE</span><span class="op">=</span><span class="st">"/scratch/PROLFQUA/r-site-library/"</span></span>
<span id="cb10-2"><a href="#cb10-2" tabindex="-1"></a><span class="ex">R</span> <span class="at">--vanilla</span> <span class="op">&lt;&lt; EOF</span></span>
<span id="cb10-3"><a href="#cb10-3" tabindex="-1"></a><span class="st">.libPaths()</span></span>
<span id="cb10-4"><a href="#cb10-4" tabindex="-1"></a><span class="st">install.packages(c("remotes","seqinr", "prozor", "logger", "arrow"), repos = "https://stat.ethz.ch/CRAN/")</span></span>
<span id="cb10-5"><a href="#cb10-5" tabindex="-1"></a><span class="st">remotes::install_gitlab("wolski/prolfquadata", host="gitlab.bfabric.org")</span></span>
<span id="cb10-6"><a href="#cb10-6" tabindex="-1"></a><span class="st">remotes::install_github("fgcz/prolfqua", build_vignettes = TRUE, dependencies = TRUE)</span></span>
<span id="cb10-7"><a href="#cb10-7" tabindex="-1"></a><span class="st">remotes::install_github("prolfqua/prolfquapp", dependencies = TRUE)</span></span>
<span id="cb10-8"><a href="#cb10-8" tabindex="-1"></a><span class="op">EOF</span></span></code></pre></div>
<p><strong>Docker</strong></p>
</div>
<div class="section level2">
<h2 id="asms-poster-streamlining-protein-differential-expression-analysis-in-core-facilities">ASMS poster: Streamlining Protein Differential Expression Analysis in Core Facilities<a class="anchor" aria-label="anchor" href="#asms-poster-streamlining-protein-differential-expression-analysis-in-core-facilities"></a>
</h2>
<div class="float">
<img src="https://github.com/prolfqua/prolfquapp/blob/master/inst/poster/prolfquapp_PosterPNG.png?raw=true" alt="prolfquapp_ASMS_poster"><div class="figcaption">prolfquapp_ASMS_poster</div>
</div>
</div>
</div>
<div class="section level1">
<h1 id="related-software">Related software<a class="anchor" aria-label="anchor" href="#related-software"></a>
</h1>
<ul>
<li>Einprot <a href="https://github.com/fmicompbio/einprot" class="external-link uri">https://github.com/fmicompbio/einprot</a>
</li>
<li>LFQAnalyst <a href="https://analyst-suite.monash-proteomics.cloud.edu.au/apps/lfq-analyst/" class="external-link uri">https://analyst-suite.monash-proteomics.cloud.edu.au/apps/lfq-analyst/</a> and <a href="https://github.com/MonashBioinformaticsPlatform/LFQ-Analyst" class="external-link uri">https://github.com/MonashBioinformaticsPlatform/LFQ-Analyst</a>
</li>
<li>POMAShiny <a href="https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1009148" class="external-link uri">https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1009148</a>
</li>
<li>MSDap <a href="https://github.com/ftwkoopmans/msdap" class="external-link uri">https://github.com/ftwkoopmans/msdap</a>
</li>
</ul>
</div>

  </main><aside class="col-md-3"><div class="links">
<h2 data-toc-skip>Links</h2>
<ul class="list-unstyled">
<li><a href="https://github.com/prolfqua/prolfquapp/" class="external-link">Browse source code</a></li>
<li><a href="https://github.com/prolfqua/prolfquapp/issues" class="external-link">Report a bug</a></li>
</ul>
</div>

<div class="license">
<h2 data-toc-skip>License</h2>
<ul class="list-unstyled">
<li><a href="LICENSE.html">Full license</a></li>
<li><small><a href="https://opensource.org/licenses/mit-license.php" class="external-link">MIT</a> + file <a href="LICENSE-text.html">LICENSE</a></small></li>
</ul>
</div>

<div class="community">
<h2 data-toc-skip>Community</h2>
<ul class="list-unstyled">
<li><a href="CODE_OF_CONDUCT.html">Code of conduct</a></li>
</ul>
</div>

<div class="citation">
<h2 data-toc-skip>Citation</h2>
<ul class="list-unstyled">
<li><a href="authors.html#citation">Citing prolfquapp</a></li>
</ul>
</div>

<div class="developers">
<h2 data-toc-skip>Developers</h2>
<ul class="list-unstyled">
<li>Witold Wolski <br><small class="roles"> Author, maintainer </small> <a href="https://orcid.org/0000-0002-6468-120X" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a>  </li>
<li>Leonardo Schwarz <br><small class="roles"> Author </small>   </li>
</ul>
</div>

<div class="dev-status">
<h2 data-toc-skip>Dev status</h2>
<ul class="list-unstyled">
<li><a href="https://doi.org/10.5281/zenodo.15845909" class="external-link"><img src="https://zenodo.org/badge/642819797.svg" alt="DOI"></a></li>
</ul>
</div>

  </aside>
</div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Witold Wolski, Leonardo Schwarz.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.2.0.</p>
</div>

    </footer>
</div>





  </body>
</html>
