<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Package index â€¢ prolfquapp</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Package index"></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">prolfquapp</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">2.0.9</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item"><a class="nav-link" href="../articles/prolfquapp.html">Get started</a></li>
<li class="active nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles"><li><a class="dropdown-item" href="../articles/Auxiliary_ExDesignSurvey.html">Auxiliary - Bioinformatics Project Meeting Agenda</a></li>
    <li><a class="dropdown-item" href="../articles/QCandSampleSize.html">Quality Control &amp; Sample Size Estimation</a></li>
  </ul></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json"></form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/prolfqua/prolfquapp/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-index">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Package index</h1>
    </div>

    <div class="section level2">
      <h2 id="all-functions">All functions<a class="anchor" aria-label="anchor" href="#all-functions"></a></h2>



      <dl></dl></div><div class="section level2">




      <dl><dt>

          <code><a href="AnnotationProcessor.html">AnnotationProcessor</a></code>

        </dt>
        <dd>AnnotationProcessor</dd>

        <dt>

          <code><a href="DEAReportGenerator.html">DEAReportGenerator</a></code>

        </dt>
        <dd>DEAReportGenerator</dd>

        <dt>

          <code><a href="DEAnalyse.html">DEAnalyse</a></code>

        </dt>
        <dd>will replace make_DEA_report</dd>

        <dt>

          <code><a href="ExternalReader.html">ExternalReader</a></code>

        </dt>
        <dd>external reader R6 class for handling external data sources</dd>

        <dt>

          <code><a href="FragPipe.html">FragPipe</a></code>

        </dt>
        <dd>Methods for reading Fragpipe outputs</dd>

        <dt>

          <code><a href="MaxQuant.html">MaxQuant</a></code>

        </dt>
        <dd>Methods for reading MaxQuant outputs</dd>

        <dt>

          <code><a href="ProcessingOptions.html">ProcessingOptions</a></code>

        </dt>
        <dd>processing options R6 class</dd>

        <dt>

          <code><a href="ProjectSpec.html">ProjectSpec</a></code>

        </dt>
        <dd>project specification R6 class</dd>

        <dt>

          <code><a href="ProjectStructure.html">ProjectStructure</a></code>

        </dt>
        <dd>keep track of folder paths and create them if needed</dd>

        <dt>

          <code><a href="ProlfquAppConfig.html">ProlfquAppConfig</a></code>

        </dt>
        <dd>R6 class representing ProlfquApp configuration</dd>

        <dt>

          <code><a href="ProteinAnnotation.html">ProteinAnnotation</a></code>

        </dt>
        <dd>Decorates LFQData with a row annotation and some protein specific functions.</dd>

        <dt>

          <code><a href="QC_generator.html">QC_generator</a></code>

        </dt>
        <dd>QC_generator</dd>

        <dt>

          <code><a href="add_RevCon.html">add_RevCon()</a></code>

        </dt>
        <dd>add REV and zz entries - used for testing</dd>

        <dt>

          <code><a href="add_contrasts_vec.html">add_contrasts_vec()</a></code>

        </dt>
        <dd>add vector of contrasts to annotation data frame</dd>

        <dt>

          <code><a href="aggregate_data.html">aggregate_data()</a></code>

        </dt>
        <dd>dataset transform data</dd>

        <dt>

          <code><a href="anndata_from_LFQData.html">anndata_from_LFQData()</a></code>

        </dt>
        <dd>convert lfqdata to anndata</dd>

        <dt>

          <code><a href="bfabric_url_builder.html">bfabric_url_builder()</a></code>

        </dt>
        <dd>build bfabric urls</dd>

        <dt>

          <code><a href="build_protein_annot.html">build_protein_annot()</a></code>

        </dt>
        <dd>build Dataset protein annot, defaults are compatible with DIANN</dd>

        <dt>

          <code><a href="capture_output.html">capture_output()</a></code>

        </dt>
        <dd>capture output of function to send it to log</dd>

        <dt>

          <code><a href="column_to_rownames.html">column_to_rownames()</a></code>

        </dt>
        <dd>convert tibble to data.frame with rownames</dd>

        <dt>

          <code><a href="compute_IBAQ_values.html">compute_IBAQ_values()</a></code>

        </dt>
        <dd>compute IBAQ values</dd>

        <dt>

          <code><a href="copy_DEA_Files.html">copy_DEA_Files()</a></code>

        </dt>
        <dd>copy Markdown and runscripts for DEA</dd>

        <dt>

          <code><a href="copy_DEA_Metabolomics_Files.html">copy_DEA_Metabolomics_Files()</a></code>

        </dt>
        <dd>copy Markdown and runscripts for DEA</dd>

        <dt>

          <code><a href="copy_docker_script.html">copy_docker_script()</a></code>

        </dt>
        <dd>copy dockerfile to run the DEA app</dd>

        <dt>

          <code><a href="copy_shell_script.html">copy_shell_script()</a></code>

        </dt>
        <dd>copy shellscript to run the DEA app</dd>

        <dt>

          <code><a href="dataset_extract_contrasts.html">dataset_extract_contrasts()</a></code>

        </dt>
        <dd>extect contrasts from dataset</dd>

        <dt>

          <code><a href="dataset_get_functions.html">dataset_get_functions()</a></code>

        </dt>
        <dd>get functions for creating datasets</dd>

        <dt>

          <code><a href="dataset_protein_annot.html">dataset_protein_annot()</a></code>

        </dt>
        <dd>Dataset protein annot</dd>

        <dt>

          <code><a href="dataset_template_BGS.html">dataset_template_BGS()</a></code>

        </dt>
        <dd>create templte dataset for BGS data</dd>

        <dt>

          <code><a href="dataset_template_FP_TMT.html">dataset_template_FP_TMT()</a></code>

        </dt>
        <dd>get dataset annotation template</dd>

        <dt>

          <code><a href="dataset_template_MAXQUANT.html">dataset_template_MAXQUANT()</a></code>

        </dt>
        <dd>create template dataset for MAXQUANT data</dd>

        <dt>

          <code><a href="dataset_template_MSSTATS.html">dataset_template_MSSTATS()</a></code>

        </dt>
        <dd>create dataset template from MSStats data.</dd>

        <dt>

          <code><a href="dataset_template_diann.html">dataset_template_diann()</a></code>

        </dt>
        <dd>create dataset template from DIANN outputs</dd>

        <dt>

          <code><a href="diann_output_to_peptide.html">diann_output_to_peptide()</a></code>

        </dt>
        <dd>Create peptide level (stripped sequences) report by aggregating Precursor abundances.</dd>

        <dt>

          <code><a href="diann_read_output.html">diann_read_output()</a></code>

        </dt>
        <dd>read DiaNN diann-output.tsv file</dd>

        <dt>

          <code><a href="diann_read_output_deprec.html">diann_read_output_deprec()</a></code>

        </dt>
        <dd>read DiaNN diann-output.tsv file</dd>

        <dt>

          <code><a href="exp2.html">exp2()</a></code>

        </dt>
        <dd>transform lfq data with x^2 - apply if non log data is needed</dd>

        <dt>

          <code><a href="extract_GN.html">extract_GN()</a></code>

        </dt>
        <dd>extract gene names from uniprot 1sp fasta.headers</dd>

        <dt>

          <code><a href="extract_contrasts.html">extract_contrasts()</a></code>

        </dt>
        <dd>extract contrast from annotation file</dd>

        <dt>

          <code><a href="feature_annotation_collapse_to_single_row.html">feature_annotation_collapse_to_single_row()</a></code>

        </dt>
        <dd>get best feature annotation.</dd>

        <dt>

          <code><a href="feature_annotation_get_best_score.html">feature_annotation_get_best_score()</a></code>

        </dt>
        <dd>get best feature annotation.</dd>

        <dt>

          <code><a href="generate_DEA_reports.html">generate_DEA_reports()</a></code>

        </dt>
        <dd>Generate differential expression analysis reports</dd>

        <dt>

          <code><a href="generate_DEA_reports2.html">generate_DEA_reports2()</a></code>

        </dt>
        <dd>will replace generate_DEA_reports</dd>

        <dt>

          <code><a href="get_BGS_files.html">get_BGS_files()</a></code>

        </dt>
        <dd>get BGS and fasta file location in folder</dd>

        <dt>

          <code><a href="get_DIANN_files.html">get_DIANN_files()</a></code>

        </dt>
        <dd>get report.tsv and fasta file location in folder</dd>

        <dt>

          <code><a href="get_FP_PSM_files.html">get_FP_PSM_files()</a></code>

        </dt>
        <dd>get psm.tsv and fasta file location in folder</dd>

        <dt>

          <code><a href="get_FP_multiSite_files.html">get_FP_multiSite_files()</a></code>

        </dt>
        <dd>get report.tsv and fasta file location in folder</dd>

        <dt>

          <code><a href="get_MQ_peptide_files.html">get_MQ_peptide_files()</a></code>

        </dt>
        <dd>get petpide.txt and fasta file location in folder</dd>

        <dt>

          <code><a href="get_MSstats_files.html">get_MSstats_files()</a></code>

        </dt>
        <dd>get petpide.txt and fasta file location in folder</dd>

        <dt>

          <code><a href="get_annot_from_fasta.html">get_annot_from_fasta()</a></code>

        </dt>
        <dd>get_annot_from_fasta</dd>

        <dt>

          <code><a href="get_config.html">get_config()</a></code>

        </dt>
        <dd>get configuration from yaml file or create default configuration</dd>

        <dt>

          <code><a href="get_dummy_files.html">get_dummy_files()</a></code>

        </dt>
        <dd>get_dummy_files</dd>

        <dt>

          <code><a href="get_formula.html">get_formula()</a></code>

        </dt>
        <dd>will generates formula</dd>

        <dt>

          <code><a href="get_mzMine_files.html">get_mzMine_files()</a></code>

        </dt>
        <dd>get mzmine fliles</dd>

        <dt>

          <code><a href="get_procfuncs.html">get_procfuncs()</a></code>

        </dt>
        <dd>Get all processing functions from all packages</dd>

        <dt>

          <code><a href="list_to_R6_app_config.html">list_to_R6_app_config()</a></code>

        </dt>
        <dd>read minimal yaml and convert to R6 object</dd>

        <dt>

          <code><a href="make_DEA_config.html">make_DEA_config()</a></code>

        </dt>
        <dd>create GRP2 configuration. Use this function if there is no Yaml Input.</dd>

        <dt>

          <code><a href="make_DEA_config_R6.html">make_DEA_config_R6()</a></code>

        </dt>
        <dd>create GRP2 configuration for differential expression analysis Use this function if there is no Yaml Input.</dd>

        <dt>

          <code><a href="make_DEA_report.html">make_DEA_report()</a></code> <code><a href="make_DEA_report.html">render_DEA()</a></code>

        </dt>
        <dd>Create DEA report in html and write data to xlsx table</dd>

        <dt>

          <code><a href="make_DEA_report2.html">make_DEA_report2()</a></code>

        </dt>
        <dd>make DEA</dd>

        <dt>

          <code><a href="make_SummarizedExperiment.html">make_SummarizedExperiment()</a></code>

        </dt>
        <dd>Convert prolfqua differential expression analysis results to SummarizedExperiment</dd>

        <dt>

          <code><a href="make_annotated_experiment.html">make_annotated_experiment()</a></code>

        </dt>
        <dd>make lfqdata with row annotation</dd>

        <dt>

          <code><a href="make_feature_annotation.html">make_feature_annotation()</a></code>

        </dt>
        <dd>get feature annotation.</dd>

        <dt>

          <code><a href="massage_CD.html">massage_CD()</a></code>

        </dt>
        <dd>massage CD output compound table.</dd>

        <dt>

          <code><a href="normalize_path.html">normalize_path()</a></code>

        </dt>
        <dd>Function to normalize paths for both Windows and Linux</dd>

        <dt>

          <code><a href="nr_tryptic_peptides.html">nr_tryptic_peptides()</a></code>

        </dt>
        <dd>Compute number of tryptic peptides</dd>

        <dt>

          <code><a href="plot_abundance_vs_percent.html">plot_abundance_vs_percent()</a></code>

        </dt>
        <dd>Plot relative protein abundance as a function of rank by abundance</dd>

        <dt>

          <code><a href="preprocess_BGS.html">preprocess_BGS()</a></code>

        </dt>
        <dd>preprocess DIANN ouput, filter by q_value and nr_peptides</dd>

        <dt>

          <code><a href="preprocess_CD.html">preprocess_CD()</a></code>

        </dt>
        <dd>load compound discoverer (CD) files</dd>

        <dt>

          <code><a href="preprocess_DIANN.html">preprocess_DIANN()</a></code>

        </dt>
        <dd>preprocess DIANN ouput, filter by q_value and nr_peptides</dd>

        <dt>

          <code><a href="preprocess_FP_PSM.html">preprocess_FP_PSM()</a></code>

        </dt>
        <dd>preprocess FP psm, filter by purity_threshold and PeptideProphetProb</dd>

        <dt>

          <code><a href="preprocess_MQ_peptide.html">preprocess_MQ_peptide()</a></code>

        </dt>
        <dd>preprocess MQ peptide</dd>

        <dt>

          <code><a href="preprocess_MSstats.html">preprocess_MSstats()</a></code>

        </dt>
        <dd>preprocess MSstats file coming from FragPipe</dd>

        <dt>

          <code><a href="preprocess_MSstats_FPDIA.html">preprocess_MSstats_FPDIA()</a></code>

        </dt>
        <dd>preprocess MSstats fragpipe</dd>

        <dt>

          <code><a href="preprocess_dummy.html">preprocess_dummy()</a></code>

        </dt>
        <dd>preprocess_dummy</dd>

        <dt>

          <code><a href="preprocess_mzMine.html">preprocess_mzMine()</a></code>

        </dt>
        <dd>preprocess mzMine input</dd>

        <dt>

          <code><a href="preprocess_software.html">preprocess_software()</a></code>

        </dt>
        <dd>collects preprocess methods for various software</dd>

        <dt>

          <code><a href="prolfqua_preprocess_functions.html">prolfqua_preprocess_functions</a></code>

        </dt>
        <dd>Mapping of software to string representations of functions and arguments Preprocess Functions Mapping</dd>

        <dt>

          <code><a href="read_BF_yamlR6.html">read_BF_yamlR6()</a></code>

        </dt>
        <dd>read yaml file and convert to R6 configuration object</dd>

        <dt>

          <code><a href="read_BGS.html">read_BGS()</a></code>

        </dt>
        <dd>get BGS and fasta file location in folder</dd>

        <dt>

          <code><a href="read_annotation.html">read_annotation()</a></code>

        </dt>
        <dd>read annotation files</dd>

        <dt>

          <code><a href="read_msstats.html">read_msstats()</a></code>

        </dt>
        <dd>read MSstats.csv files and rollup to ProteinSequence level.</dd>

        <dt>

          <code><a href="read_table_data.html">read_table_data()</a></code>

        </dt>
        <dd>read dataset file in csv, tsv or xlsx format</dd>

        <dt>

          <code><a href="read_yaml_deprec.html">read_yaml_deprec()</a></code>

        </dt>
        <dd>read yaml file</dd>

        <dt>

          <code><a href="sanitize_grouping_var.html">sanitize_grouping_var()</a></code>

        </dt>
        <dd>Sanitize grouping variable in annotation file</dd>

        <dt>

          <code><a href="set_lib_path.html">set_lib_path()</a></code>

        </dt>
        <dd>set library path with logging</dd>

        <dt>

          <code><a href="set_list_to_R6.html">set_list_to_R6()</a></code>

        </dt>
        <dd>set arguments in list config to r6obj</dd>

        <dt>

          <code><a href="sim_data_protAnnot.html">sim_data_protAnnot()</a></code>

        </dt>
        <dd>simulate peptdata and fitting protein annotation for testing</dd>

        <dt>

          <code><a href="sync_opt_config.html">sync_opt_config()</a></code>

        </dt>
        <dd>Synchronize opt and config</dd>

        <dt>

          <code><a href="tidy_FragPipe_psm.html">tidy_FragPipe_psm()</a></code>

        </dt>
        <dd>read psm.tsv produced by FragPipe and convert into long format</dd>

        <dt>

          <code><a href="tidy_FragPipe_psm_V2.html">tidy_FragPipe_psm_V2()</a></code>

        </dt>
        <dd>read psm.tsv produced by FragPipe and convert into long format</dd>

        <dt>

          <code><a href="tidy_mzMineFeatures.html">tidy_mzMineFeatures()</a></code>

        </dt>
        <dd>convert mzmine features to tidy table</dd>

        <dt>

          <code><a href="transform_lfqdata.html">transform_lfqdata()</a></code>

        </dt>
        <dd>transform lfq data using robscale, vsn or log2, Assumes that data is not transformed (still needs log2 transformation)</dd>

        <dt>

          <code><a href="writeLinesPaired.html">writeLinesPaired()</a></code>

        </dt>
        <dd>nice plot for paired analysis</dd>

        <dt>

          <code><a href="write_DEA.html">write_DEA()</a></code>

        </dt>
        <dd>Write differential expression analysis results</dd>

        <dt>

          <code><a href="write_DEA_all.html">write_DEA_all()</a></code>

        </dt>
        <dd>Generate differential expression analysis reports</dd>

        <dt>

          <code><a href="write_annotation_file.html">write_annotation_file()</a></code>

        </dt>
        <dd>Write dataset to file in csv, tsv, or xlsx format</dd>

        <dt>

          <code><a href="write_index_html.html">write_index_html()</a></code>

        </dt>
        <dd>write index.html file with links to all relevant files:</dd>

        <dt>

          <code><a href="zipdir_name.html">zipdir_name()</a></code>

        </dt>
        <dd>Generate zip directory name based on project information and date</dd>
      </dl></div>
  </main></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Witold Wolski, Leonardo Schwarz.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.2.0.</p>
</div>

    </footer></div>





  </body></html>

