<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Auxiliary - Bioinformatics Project Meeting Agenda • prolfquapp</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Auxiliary - Bioinformatics Project Meeting Agenda">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">prolfquapp</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">2.0.10</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../articles/prolfquapp.html">Get started</a></li>
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="active nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="../articles/Auxiliary_ExDesignSurvey.html">Auxiliary - Bioinformatics Project Meeting Agenda</a></li>
    <li><a class="dropdown-item" href="../articles/QCandSampleSize.html">Quality Control &amp; Sample Size Estimation</a></li>
  </ul>
</li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/prolfqua/prolfquapp/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Auxiliary - Bioinformatics Project Meeting Agenda</h1>
                        <h4 data-toc-skip class="author">FGCZ -
(Draft)</h4>
            
            <h4 data-toc-skip class="date">25 February, 2026</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/prolfqua/prolfquapp/blob/master/vignettes/Auxiliary_ExDesignSurvey.Rmd" class="external-link"><code>vignettes/Auxiliary_ExDesignSurvey.Rmd</code></a></small>
      <div class="d-none name"><code>Auxiliary_ExDesignSurvey.Rmd</code></div>
    </div>

    
    
<p>Project Nr :</p>
<p>Order Id :</p>
<p>Participants:</p>
<ul>
<li>User :</li>
<li>Bioinformatician :</li>
<li>Coach :</li>
</ul>
<p>When?</p>
<ul class="task-list">
<li><label><input type="checkbox">before start of
experiment</label></li>
<li><label><input type="checkbox">after QC sample
preparation</label></li>
<li><label><input type="checkbox">after main experiment sample
preparation</label></li>
</ul>
<p>Comments :</p>
<p><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mspace width="0.222em"></mspace><annotation encoding="application/x-tex">~</annotation></semantics></math></p>
<p><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mspace width="0.222em"></mspace><annotation encoding="application/x-tex">~</annotation></semantics></math></p>
<p><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mspace width="0.222em"></mspace><annotation encoding="application/x-tex">~</annotation></semantics></math></p>
<p><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mspace width="0.222em"></mspace><annotation encoding="application/x-tex">~</annotation></semantics></math></p>
<div class="section level2">
<h2 id="overview-of-the-analysis">Overview of the analysis<a class="anchor" aria-label="anchor" href="#overview-of-the-analysis"></a>
</h2>
<p>In label-free quantification (LFQ) experiments, we measure relative
protein changes among groups of subjects. Before the main experiment, a
quality control (QC) experiment is performed. We use the data from the
QC to estimate the <em>within-group</em> variance and to determine the
number of samples in the main experiment. The measurements of the main
experiment are modeled using linear models, mixed-effects linear models,
which can be thought of as an extension of the t-test to more than two
groups. These estimates are used to perform Gene Set Enrichment Analysis
and Overrepresentation analysis using WebGestaltR and sigora. The
meeting aims to obtain and document all the information relevant to
perform a statistically sound LFQ data analysis.</p>
<div class="section level3">
<h3 id="type-of-proteomics-experiment">Type of proteomics experiment<a class="anchor" aria-label="anchor" href="#type-of-proteomics-experiment"></a>
</h3>
<ul class="task-list">
<li>
<label><input type="checkbox">data dependent aquisition (DDA)
experiment (robust method, typically used for affinity purification
experiments)</label>
<ul>
<li>Pro : robust, can always be applied</li>
<li>Con : fewer ids and quant values than with other methods</li>
</ul>
</li>
<li>
<label><input type="checkbox">data independent aquisition (DIA)
experiment (improved sensitivity)</label>
<ul>
<li>Pro : improved sensitivity, works better if more samples are
measured.</li>
<li>Con : needs at least 8 samples</li>
</ul>
</li>
<li>
<label><input type="checkbox">tandem mass tag (TMT)
experiment</label>
<ul>
<li>Pro : improves sensitivity (more proteins identified) by
fractionating samples</li>
<li>Con : fold change compression</li>
</ul>
</li>
</ul>
</div>
<div class="section level3">
<h3 id="sample-modifications">Sample modifications<a class="anchor" aria-label="anchor" href="#sample-modifications"></a>
</h3>
<p>Which fixed an variable modfications are used:</p>
<p>Variable Modifictions: - [ ] acetyl - [ ] biotin - [ ] deamidation -
[ ] dimethyl - [ ] formylation - [ ] methylation - [ ] O - Glycosylation
HexNAc</p>
</div>
</div>
<div class="section level2">
<h2 id="aim-of-the-meeting">Aim of the meeting<a class="anchor" aria-label="anchor" href="#aim-of-the-meeting"></a>
</h2>
<p>The goal of the meeting is this protocol specifying:</p>
<ul>
<li>which protein database to use</li>
<li>which and how many samples to use for the QC</li>
<li>determine all the parameters for the sample size estimation</li>
<li>the design of your main experiment</li>
<li>all the names of the factors and factor levels</li>
<li>specify the hypothesis to be tested</li>
</ul>
<p>The samples should be annotated with the names specified in the
protocol in the b-fabric system. These names will also appear in the
result tables and visualizations.</p>
<p>We perform the data analysis according to this protocol. Changing the
analysis protocol (e.g. adding contrasts or blocking factors) after the
analysis was delivered will incur time delays and additional costs.</p>
</div>
<div class="section level2">
<h2 id="genomics-analysis">Genomics Analysis<a class="anchor" aria-label="anchor" href="#genomics-analysis"></a>
</h2>
<p>Did you perform a genomics experiment on the same samples?</p>
<ul class="task-list">
<li><label><input type="checkbox">no</label></li>
<li><label><input type="checkbox">yes</label></li>
</ul>
<p>What type of experiment?</p>
<p><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mspace width="0.222em"></mspace><annotation encoding="application/x-tex">~</annotation></semantics></math></p>
<p><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mspace width="0.222em"></mspace><annotation encoding="application/x-tex">~</annotation></semantics></math></p>
<ul>
<li>Was it run at the FGCZ?</li>
<li>project Id:</li>
<li>order Id:</li>
<li>Bioinformatician :</li>
<li>Are you interested in Proteomics Genomics data integration?</li>
</ul>
<p><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mspace width="0.222em"></mspace><annotation encoding="application/x-tex">~</annotation></semantics></math></p>
<p><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mspace width="0.222em"></mspace><annotation encoding="application/x-tex">~</annotation></semantics></math></p>
<p>Note:</p>
<ul>
<li>NGS data can be used to create improved protein identification
databases.</li>
<li>If NGS - proteomics data integration is planned, compatible
statistical analysis should be applied to the NGS and Proteomics
data.</li>
</ul>
</div>
<div class="section level2">
<h2 id="protein-sequence-database-and-downstream-analysis-tools">Protein sequence database and downstream analysis tools<a class="anchor" aria-label="anchor" href="#protein-sequence-database-and-downstream-analysis-tools"></a>
</h2>
<p>Which sequence database should be used for protein
identification?</p>
<p>Which organism?</p>
<ul>
<li><label><input type="checkbox">homo sapiens</label></li>
<li><label><input type="checkbox">mus musculus</label></li>
<li><label><input type="checkbox">yeast</label></li>
<li><label><input type="checkbox">zebra-fish</label></li>
<li>other :</li>
</ul>
<p>Should additional protein sequences be added to the database?</p>
<p><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mspace width="0.222em"></mspace><annotation encoding="application/x-tex">~</annotation></semantics></math></p>
<p><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mspace width="0.222em"></mspace><annotation encoding="application/x-tex">~</annotation></semantics></math></p>
<p>Contaminant list?</p>
<ul>
<li><label><input type="checkbox">Complete</label></li>
<li><label><input type="checkbox">Without human
contaminants</label></li>
<li>other :</li>
</ul>
<p>Which downstream analysis tools will be used?</p>
<ul>
<li><label><input type="checkbox"><a href="http://www.webgestalt.org/" target="_blank" class="external-link">WebGestalt</a> (WEB-based GEne SeT AnaLysis
Toolkit)</label></li>
<li><label><input type="checkbox"><a href="https://string-db.org/" target="_blank" class="external-link">string-db</a> (Known and predicted protein-protein
interactions)</label></li>
<li>other :</li>
</ul>
<p><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mspace width="0.222em"></mspace><annotation encoding="application/x-tex">~</annotation></semantics></math></p>
<p>Note:</p>
<ul>
<li><p>Typically we are using protein databases from
uniprot.org</p></li>
<li><p>Ideally the identifiers in the protein database are compatible
with the downstream analysis tool you intend to use. For instance
<em>WebGestalt</em>, although it recognizes many types of identifiers:
e.g., Uniprot Swiss-Prot, Gene Symbols, NCBI Entrez gene, it does not
work with Uniprot Treml identifiers.</p></li>
</ul>
</div>
<div class="section level2">
<h2 id="effect-size-and-size-of-test">Effect size and size of test<a class="anchor" aria-label="anchor" href="#effect-size-and-size-of-test"></a>
</h2>
<p>These parameters are crucial for the QC experiment and will be
assessed again when protein variance estimates from the QC experiment
are available. They also will be used when visualizing the results in
the final report (e.g., Volcano plots) or when running
overrepresentation analysis (ORA).</p>
<div class="section level3">
<h3 id="effect-size">Effect size<a class="anchor" aria-label="anchor" href="#effect-size"></a>
</h3>
<p>What is the smallest effect size (i.e., fold change in protein
abundance) of biological relevance, you are interested in detecting?
Typically fold changes are
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mn>1.5</mn><annotation encoding="application/x-tex">1.5</annotation></semantics></math>,
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mn>2</mn><annotation encoding="application/x-tex">2</annotation></semantics></math>,
or
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mn>4</mn><annotation encoding="application/x-tex">4</annotation></semantics></math>.
This information is needed to determine the sample size as well as to
biologically interpret the data.</p>
<p>The number of samples you need to use highly depends on the effect
size you want to detect. Detecting small effect sizes requires you to
measure more samples. On the other hand, the statistical significance of
a fold change does not imply biological relevance. Consequently, do not
measure more samples than is required.</p>
<ul>
<li><label><input type="checkbox">1.5</label></li>
<li><label><input type="checkbox">2</label></li>
<li><label><input type="checkbox">4</label></li>
<li>other :</li>
</ul>
</div>
<div class="section level3">
<h3 id="size-of-the-test5">Size of the test<a class="footnote-ref" tabindex="0" data-bs-toggle="popover" data-bs-content="&lt;p&gt;In statistics, the size of a test is the probability of
falsely rejecting the null hypothesis.&lt;/p&gt;"><sup>1</sup></a><a class="anchor" aria-label="anchor" href="#size-of-the-test5"></a>
</h3>
<p>Typically a difference between two conditions is considered to be
significant if the p-Value is less than
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mn>0.05</mn><annotation encoding="application/x-tex">0.05</annotation></semantics></math>.
For some applications a less significant p-Value can be used (e.g.,
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mn>0.1</mn><annotation encoding="application/x-tex">0.1</annotation></semantics></math>)
for others a smaller p-Value might be required.</p>
<ul>
<li><label><input type="checkbox">0.01</label></li>
<li><label><input type="checkbox">0.05</label></li>
<li><label><input type="checkbox">0.1</label></li>
<li>other :</li>
</ul>
</div>
<div class="section level3">
<h3 id="power-of-test">Power of test<a class="anchor" aria-label="anchor" href="#power-of-test"></a>
</h3>
<p>The power of a binary hypothesis test is the probability that the
test rejects the null hypothesis (H0) when a specific alternative
hypothesis (H1) is true <a href="https://en.wikipedia.org/wiki/Power_(statistics)" class="external-link">Wikipedia</a></p>
<ul>
<li><label><input type="checkbox">0.8</label></li>
<li>other :</li>
</ul>
</div>
<div class="section level3">
<h3 id="does-the-variance-within-groups-differ">Does the variance within groups differ?<a class="anchor" aria-label="anchor" href="#does-the-variance-within-groups-differ"></a>
</h3>
<p>Is the variance within the groups the same or different? E.g., do you
expect some groups with larger within group differences?</p>
<ul class="task-list">
<li><label><input type="checkbox">yes</label></li>
<li><label><input type="checkbox">no</label></li>
</ul>
<p>If yes, which of the group’s might have the highest within-group
variance? (Some treatment might have heterogeneous outcomes.)</p>
<p><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mspace width="0.222em"></mspace><annotation encoding="application/x-tex">~</annotation></semantics></math><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mspace width="0.222em"></mspace><annotation encoding="application/x-tex">~</annotation></semantics></math></p>
<p>If possible, use samples from this group for the QC experiment, which
will enable you to get a conservative estimate of the required samples
sizes. If the variance in one group is substantially higher, the power
of the tests can be improved by allocating more samples to this
group.</p>
</div>
<div class="section level3">
<h3 id="difference-in-protein-variances">Difference in protein variances<a class="anchor" aria-label="anchor" href="#difference-in-protein-variances"></a>
</h3>
<p>From the QC experiment, we can estimate for each protein a
within-group variance. For some proteins, we observe more variation
within a group compared to other proteins.</p>
<p>Are there proteins or protein groups of particular interest?</p>
<p>Provide protein IDs:</p>
<p>Do you want to obtain significant p-Values for proteins with large
within-group variances, or would you be satisfied with getting
significances for the
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>X</mi><mi>%</mi></mrow><annotation encoding="application/x-tex">X\%</annotation></semantics></math>
of low variance proteins?</p>
<ul>
<li><label><input type="checkbox"><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mn>50</mn><mi>%</mi></mrow><annotation encoding="application/x-tex">50\%</annotation></semantics></math></label></li>
<li><label><input type="checkbox"><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mn>70</mn><mi>%</mi></mrow><annotation encoding="application/x-tex">70\%</annotation></semantics></math></label></li>
<li>other :</li>
</ul>
<p>Note:</p>
<ul>
<li>High abundant proteins will exhibit smaller technical variability
compared with low abundant proteins.</li>
</ul>
</div>
</div>
<div class="section level2">
<h2 id="designs">Designs<a class="anchor" aria-label="anchor" href="#designs"></a>
</h2>
<p>This section covers the most frequently used designs.</p>
<ul class="task-list">
<li><p><label><input type="checkbox">Parallel Group Design - Evaluates
a single factor<a class="footnote-ref" tabindex="0" data-bs-toggle="popover" data-bs-content="&lt;p&gt;A parallel design, also called a parallel group study,
compares two or more treatments. Participants are randomly assigned to
either group, treatments are administered, and then the results are
compared (statistics how to).&lt;/p&gt;"><sup>2</sup></a></label></p></li>
<li><p><label><input type="checkbox">Factorial Design - Evaluates
multiple factors simultaneously</label></p></li>
</ul>
<div class="section level3">
<h3 id="parallel-group-design">Parallel Group Design<a class="anchor" aria-label="anchor" href="#parallel-group-design"></a>
</h3>
<p>How many groups?</p>
<ul class="task-list">
<li><label><input type="checkbox">2</label></li>
<li><label><input type="checkbox">more. How many?</label></li>
</ul>
<p>Factor name (<em>camel case</em><a class="footnote-ref" tabindex="0" data-bs-toggle="popover" data-bs-content="&lt;p&gt;&lt;em&gt;camel case&lt;/em&gt; - a typographical convention in
which an initial capital is used for the first letter of a word-forming
the second element of a closed compound, e.g., PayPal, iPhone,
MasterCard.&lt;/p&gt;"><sup>3</sup></a> e.g., Treatment) :</p>
<p>Level names (<em>camel case</em> e.g. Control, Diet, Creme):</p>
</div>
<div class="section level3">
<h3 id="factorial-design8">Factorial Design<a class="footnote-ref" tabindex="0" data-bs-toggle="popover" data-bs-content="&lt;p&gt;In statistics, a full factorial experiment is an
experiment whose design consists of two or more factors, each with
discrete possible values or “levels”, and whose experimental units take
on all possible combinations of these levels across all such factors.
Such an experiment allows the investigator to study the effect of each
factor on the response variable, as well as the effects of interactions
between factors on the response variable.&lt;/p&gt;"><sup>4</sup></a><a class="anchor" aria-label="anchor" href="#factorial-design8"></a>
</h3>
<p>Applies when studying how two or more factors influence protein
expression, e.g.,
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>K</mi><mi>n</mi><mi>o</mi><mi>c</mi><mi>k</mi><mi>o</mi><mi>u</mi><mi>t</mi></mrow><annotation encoding="application/x-tex">Knockout</annotation></semantics></math>
and
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>T</mi><mi>r</mi><mi>e</mi><mi>a</mi><mi>t</mi><mi>m</mi><mi>e</mi><mi>n</mi><mi>t</mi></mrow><annotation encoding="application/x-tex">Treatment</annotation></semantics></math>.</p>
<ul>
<li>How many factors? Typically 2 rarely 3.</li>
<li>How many levels per factor? Typically 2 or 3 per factor.</li>
</ul>
<table class="table">
<colgroup>
<col width="37%">
<col width="25%">
<col width="37%">
</colgroup>
<thead><tr class="header">
<th align="left">Factor Name</th>
<th>Number levels</th>
<th align="center">level names
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mspace width="0.222em"></mspace><mspace width="0.222em"></mspace><mspace width="0.222em"></mspace><mspace width="0.222em"></mspace><mspace width="0.222em"></mspace><mspace width="0.222em"></mspace><mspace width="0.222em"></mspace><mspace width="0.222em"></mspace><mspace width="0.222em"></mspace><mspace width="0.222em"></mspace><mspace width="0.222em"></mspace><mspace width="0.222em"></mspace><mspace width="0.222em"></mspace><mspace width="0.222em"></mspace><mspace width="0.222em"></mspace><mspace width="0.222em"></mspace><mspace width="0.222em"></mspace><mspace width="0.222em"></mspace><mspace width="0.222em"></mspace><mspace width="0.222em"></mspace><mspace width="0.222em"></mspace><mspace width="0.222em"></mspace><mspace width="0.222em"></mspace><mspace width="0.222em"></mspace><mspace width="0.222em"></mspace><mspace width="0.222em"></mspace><mspace width="0.222em"></mspace><mspace width="0.222em"></mspace><mspace width="0.222em"></mspace><mspace width="0.222em"></mspace><mspace width="0.222em"></mspace><mspace width="0.222em"></mspace><mspace width="0.222em"></mspace></mrow><annotation encoding="application/x-tex">~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~</annotation></semantics></math>
</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left"><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mspace width="0.222em"></mspace><annotation encoding="application/x-tex">~</annotation></semantics></math></td>
<td></td>
<td align="center"></td>
</tr>
<tr class="even">
<td align="left"><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mspace width="0.222em"></mspace><annotation encoding="application/x-tex">~</annotation></semantics></math></td>
<td></td>
<td align="center"></td>
</tr>
<tr class="odd">
<td align="left"><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mspace width="0.222em"></mspace><annotation encoding="application/x-tex">~</annotation></semantics></math></td>
<td></td>
<td align="center"></td>
</tr>
<tr class="even">
<td align="left"><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mspace width="0.222em"></mspace><annotation encoding="application/x-tex">~</annotation></semantics></math></td>
<td></td>
<td align="center"></td>
</tr>
</tbody>
</table>
<ul>
<li>Provide computer-friendly names of the factors (grouping variables)
in <em>camel case</em>.</li>
<li>Provide names of levels for each grouping variable.</li>
</ul>
<p>Example:</p>
<table class="table">
<colgroup>
<col width="37%">
<col width="25%">
<col width="37%">
</colgroup>
<thead><tr class="header">
<th align="left">Factor Name</th>
<th>Number levels</th>
<th align="center">level names
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mspace width="0.222em"></mspace><mspace width="0.222em"></mspace><mspace width="0.222em"></mspace><mspace width="0.222em"></mspace><mspace width="0.222em"></mspace><mspace width="0.222em"></mspace><mspace width="0.222em"></mspace><mspace width="0.222em"></mspace><mspace width="0.222em"></mspace><mspace width="0.222em"></mspace><mspace width="0.222em"></mspace><mspace width="0.222em"></mspace><mspace width="0.222em"></mspace><mspace width="0.222em"></mspace><mspace width="0.222em"></mspace><mspace width="0.222em"></mspace><mspace width="0.222em"></mspace><mspace width="0.222em"></mspace><mspace width="0.222em"></mspace><mspace width="0.222em"></mspace><mspace width="0.222em"></mspace><mspace width="0.222em"></mspace><mspace width="0.222em"></mspace><mspace width="0.222em"></mspace><mspace width="0.222em"></mspace><mspace width="0.222em"></mspace><mspace width="0.222em"></mspace><mspace width="0.222em"></mspace><mspace width="0.222em"></mspace><mspace width="0.222em"></mspace><mspace width="0.222em"></mspace><mspace width="0.222em"></mspace><mspace width="0.222em"></mspace></mrow><annotation encoding="application/x-tex">~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~</annotation></semantics></math>
</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">Treatment</td>
<td>3</td>
<td align="center">Control, Diet, Creme</td>
</tr>
<tr class="even">
<td align="left">AgeGroup</td>
<td>2</td>
<td align="center">young, old</td>
</tr>
</tbody>
</table>
</div>
<div class="section level3">
<h3 id="repeated-measurements">Repeated measurements<a class="anchor" aria-label="anchor" href="#repeated-measurements"></a>
</h3>
<p>Applies when measurements are repeated on the same subject, e.g. over
time <a class="footnote-ref" tabindex="0" data-bs-toggle="popover" data-bs-content="&lt;p&gt;Repeated measures design is a research design that
involves multiple measures of the same variable taken on the same or
matched subjects either under different conditions or over two or more
time periods. For instance, repeated measurements are collected in a
longitudinal study in which change over time is assessed (Wikipedia).&lt;/p&gt;"><sup>5</sup></a>.</p>
<ul class="task-list">
<li><label><input type="checkbox">Yes</label></li>
<li><label><input type="checkbox">No</label></li>
</ul>
<p>If <em>yes</em> please specify a computer-friendly factor name using
<em>camel case</em>: e.g., PatientId.</p>
<p>Name of the factor :</p>
<p>The subject identifiers can contain upper and lower case letters and
numbers but avoid special characters and white spaces. Neither the
factor names or the levels of the blocking factors will be shown in the
results or figures generated.</p>
</div>
<div class="section level3">
<h3 id="blocking-factors">Blocking factors<a class="anchor" aria-label="anchor" href="#blocking-factors"></a>
</h3>
<p>A blocking factor is some variable that affects an experimental
outcome but is itself of no interest, e.g., sample batch <a class="footnote-ref" tabindex="0" data-bs-toggle="popover" data-bs-content='&lt;p&gt;A blocking factor is a factor used to create blocks. It
is some variable that has an effect on an experimental outcome, but is
itself of no interest. Blocking factors vary wildly depending on the
experiment. For example: in human studies age or gender are often used
as blocking factors. Blocking factors can be included into the model to
describe the variance of the data &lt;a href="https://www.statisticshowto.datasciencecentral.com/blocking-factor/" class="external-link"&gt;data
science central&lt;/a&gt;.&lt;/p&gt;'><sup>6</sup></a>. Including
blocking-factors can improve the statistical model.</p>
<table class="table">
<thead><tr class="header">
<th align="left">Factor Name</th>
<th>Number levels</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left"><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mspace width="0.222em"></mspace><annotation encoding="application/x-tex">~</annotation></semantics></math></td>
<td></td>
</tr>
<tr class="even">
<td align="left"><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mspace width="0.222em"></mspace><annotation encoding="application/x-tex">~</annotation></semantics></math></td>
<td></td>
</tr>
<tr class="odd">
<td align="left"><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mspace width="0.222em"></mspace><annotation encoding="application/x-tex">~</annotation></semantics></math></td>
<td></td>
</tr>
<tr class="even">
<td align="left"><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mspace width="0.222em"></mspace><annotation encoding="application/x-tex">~</annotation></semantics></math></td>
<td></td>
</tr>
</tbody>
</table>
</div>
<div class="section level3">
<h3 id="further-information">Further information<a class="anchor" aria-label="anchor" href="#further-information"></a>
</h3>
<p><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mspace width="0.222em"></mspace><annotation encoding="application/x-tex">~</annotation></semantics></math></p>
<p><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mspace width="0.222em"></mspace><annotation encoding="application/x-tex">~</annotation></semantics></math></p>
<p><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mspace width="0.222em"></mspace><annotation encoding="application/x-tex">~</annotation></semantics></math></p>
<p><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mspace width="0.222em"></mspace><annotation encoding="application/x-tex">~</annotation></semantics></math></p>
<p><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mspace width="0.222em"></mspace><annotation encoding="application/x-tex">~</annotation></semantics></math></p>
<p><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mspace width="0.222em"></mspace><annotation encoding="application/x-tex">~</annotation></semantics></math></p>
</div>
</div>
<div class="section level2">
<h2 id="contrasts---hypothesis-to-be-tested">Contrasts - Hypothesis to be tested<a class="anchor" aria-label="anchor" href="#contrasts---hypothesis-to-be-tested"></a>
</h2>
<p>If there are more than two groups, specify the hypothesis to be
tested (comparison) using the factor and factor level names.</p>
<p><strong><em>Contrasts to be computed:</em></strong></p>
<p><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mspace width="0.222em"></mspace><annotation encoding="application/x-tex">~</annotation></semantics></math></p>
<p><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mspace width="0.222em"></mspace><annotation encoding="application/x-tex">~</annotation></semantics></math></p>
<p><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mspace width="0.222em"></mspace><annotation encoding="application/x-tex">~</annotation></semantics></math></p>
<p><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mspace width="0.222em"></mspace><annotation encoding="application/x-tex">~</annotation></semantics></math></p>
<p><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mspace width="0.222em"></mspace><annotation encoding="application/x-tex">~</annotation></semantics></math></p>
<p><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mspace width="0.222em"></mspace><annotation encoding="application/x-tex">~</annotation></semantics></math></p>
<p><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mspace width="0.222em"></mspace><annotation encoding="application/x-tex">~</annotation></semantics></math></p>
<p><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mspace width="0.222em"></mspace><annotation encoding="application/x-tex">~</annotation></semantics></math></p>
<p><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mspace width="0.222em"></mspace><annotation encoding="application/x-tex">~</annotation></semantics></math></p>
<p>Examples for parallel Group design (only one factor):</p>
<ul>
<li>Diet - Control</li>
<li>Creme - Control</li>
</ul>
<p>Example for factorial designs:</p>
<ul>
<li>Treatment_Control - Treatment_Creme</li>
<li>AgeGroup_young - AgeGroup_old</li>
<li>Treatment_Control:AgeGroup_young -
Treatment_Control:AgeGroup_old</li>
</ul>
<p>Note:</p>
<ul>
<li><p>If one group is used in more comparisons than all other groups,
it will benefit the power of the test <a class="footnote-ref" tabindex="0" data-bs-toggle="popover" data-bs-content='&lt;p&gt;The power of a binary hypothesis test is the probability
that the test rejects the null hypothesis (H0) when a specific
alternative hypothesis (H1) is true. The statistical power ranges from 0
to 1, and as statistical power increases, the probability of making a
type II error (wrongly failing to reject the null hypothesis) decreases.
&lt;a href="https://en.wikipedia.org/wiki/Power_(statistics)" class="external-link"&gt;Wikipedia&lt;/a&gt;&lt;/p&gt;'><sup>7</sup></a> to allocate more
samples to this group, since increasing the number of degrees of freedom
in this group will benefit many comparisons.</p></li>
<li><p><strong>Limit the number of hypotheses you test</strong>.
Multiplicity - each comparison is a test of hypothesis. Therefore, it is
required to perform a p-Value adjustment. For instance, using the
Bonferroni correction, the obtained p-Values are multiplied by the
number of the hypothesis tested. Given two comparisons, a p-Value of
0.03 after Bonferroni adjustment is
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mn>0.06</mn><annotation encoding="application/x-tex">0.06</annotation></semantics></math>,
i.e., not significant given a test size of
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mn>0.05</mn><annotation encoding="application/x-tex">0.05</annotation></semantics></math>.
Therefore, do not test all possible hypothesis! For instance, if
including the contrast <em>Diet - Creme</em>, it is required to multiply
the p-Values with the factor
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mn>3</mn><annotation encoding="application/x-tex">3</annotation></semantics></math>.
i.e., a p-Value of
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mn>0.2</mn><annotation encoding="application/x-tex">0.2</annotation></semantics></math>
after Bonferroni adjustment is
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mn>0.6</mn><annotation encoding="application/x-tex">0.6</annotation></semantics></math>
- i.e., not significant.</p></li>
</ul>
<ul>
<li><p>Factorial designs generate many interactions (e.g.,
Treatment_Control:AgeGroup_young). A
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mn>3</mn><mo>×</mo><mn>2</mn></mrow><annotation encoding="application/x-tex">3\times 2</annotation></semantics></math>
factorial design generates
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mn>6</mn><annotation encoding="application/x-tex">6</annotation></semantics></math>
groups (unique combinations of factors) which makes it possible to
specify many contrasts. Limit the number of hypotheses ideally to few
(one or two) per experiment. It is required to adjust the p-Values based
on the number of hypotheses examined.</p></li>
<li><p>Typically in factorial designs, tests for differences among main
effects (differences between levels within a single factor) and
second-order interactions of the factors.</p></li>
<li><p><em>main effects</em> : Treatment_Control -
Treatment_Diet</p></li>
<li><p><em>Test for interaction</em> : (Treatment_Control:AgeGroup_young
- Treatment_Control:AgeGroup_old) - (Treatment_Diet:AgeGroup_young -
Treatment_Diet:AgeGroup_old)</p></li>
</ul>
<div class="section level3">
<h3 id="more-designs">More designs<a class="anchor" aria-label="anchor" href="#more-designs"></a>
</h3>
<ul>
<li>Time Series, e.g., growth curves - require at least 4-5 different
well-designed time-points. When less than 4-time points are measured
than the time points are typically treated as factors and can be
analyzed using our standard pipeline for factorial designs.</li>
</ul>
<p>We might be able to support you with some other hypothesis tests,
depending on our availability.</p>
<ul>
<li>Dose-response curves</li>
<li>Survival analysis</li>
</ul>
</div>
</div>
<div class="section level2">
<h2 id="other-topics">Other topics<a class="anchor" aria-label="anchor" href="#other-topics"></a>
</h2>
<p>Example: Normalization is applied to remove systematic differences in
protein abundance due to different sample concentrations, or different
amount of sample loaded on column. Normalization is important, so that
true differentially expressed proteins can be detected.</p>
<p><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mspace width="0.222em"></mspace><annotation encoding="application/x-tex">~</annotation></semantics></math></p>
<p><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mspace width="0.222em"></mspace><annotation encoding="application/x-tex">~</annotation></semantics></math></p>
<p><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mspace width="0.222em"></mspace><annotation encoding="application/x-tex">~</annotation></semantics></math></p>
<p><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mspace width="0.222em"></mspace><annotation encoding="application/x-tex">~</annotation></semantics></math></p>
<p><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mspace width="0.222em"></mspace><annotation encoding="application/x-tex">~</annotation></semantics></math></p>
<p><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mspace width="0.222em"></mspace><annotation encoding="application/x-tex">~</annotation></semantics></math></p>
<p><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mspace width="0.222em"></mspace><annotation encoding="application/x-tex">~</annotation></semantics></math></p>
<p><math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mspace width="0.222em"></mspace><annotation encoding="application/x-tex">~</annotation></semantics></math></p>
<p>Date:</p>
</div>

  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Witold Wolski, Leonardo Schwarz.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.2.0.</p>
</div>

    </footer>
</div>





  </body>
</html>
