<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>prolfquapp - an R package to report differential expression analysis results. • prolfquapp</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="prolfquapp - an R package to report differential expression analysis results.">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">prolfquapp</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">2.0.9</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="active nav-item"><a class="nav-link" href="../articles/prolfquapp.html">Get started</a></li>
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="../articles/Auxiliary_ExDesignSurvey.html">Auxiliary - Bioinformatics Project Meeting Agenda</a></li>
    <li><a class="dropdown-item" href="../articles/QCandSampleSize.html">Quality Control &amp; Sample Size Estimation</a></li>
  </ul>
</li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/prolfqua/prolfquapp/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>prolfquapp - an R package to report differential expression analysis results.</h1>
            
            <h4 data-toc-skip class="date">2022-12-21</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/prolfqua/prolfquapp/blob/master/vignettes/prolfquapp.Rmd" class="external-link"><code>vignettes/prolfquapp.Rmd</code></a></small>
      <div class="d-none name"><code>prolfquapp.Rmd</code></div>
    </div>

    
    
<div class="section level3">
<h3 id="r-markdown">R Markdown<a class="anchor" aria-label="anchor" href="#r-markdown"></a>
</h3>
<p>To ease the usage barriers of the <em>R</em> package to users not
familiar with statistics and <em>R</em> programming, we developed an
application based on the <em>prolfqua</em> package into our data
management platform <a href="https://fgcz-bfabric.uzh.ch/" class="external-link">B-Fabric</a>
<span class="citation">[@bfabric;@bfabricjib2022]</span>. The B-Fabric
system runs a computing infrastructure controlled by a local resource
management system that supports cloud-bursting <span class="citation">[@VMMAD]</span>. This integration enables users to
select the input data and basic settings in a graphical user interface
(GUI).</p>
<p>In this GUI application, we determine the model formula, and which
comparisons to compute from the sample annotation provided in tabular
form, similarly to <em>SAINTexpress</em> or <em>MSstats</em>. However,
this application can analyze only parallel group designs with and
without repeated measurements or factorial designs without interactions.
The user receives a report including input files, the <em>R</em>
markdown file, and <em>R</em> scripts necessary to replicate the
analysis using their in-house <em>R</em> installation. In this way,
<em>prolfqua</em> and <a href="https://fgcz-bfabric.uzh.ch/" class="external-link">B-Fabric</a> help scientists to meet
requirements from funding agencies, journals, and academic institutions
while publishing their data according to the FAIR <span class="citation">[@FAIR]</span> data principles. We are working on
creating a shiny stand-alone application with the described
functionality and making it available soon.</p>
</div>
<div class="section level2">
<h2 id="describe-inputs">Describe inputs<a class="anchor" aria-label="anchor" href="#describe-inputs"></a>
</h2>
</div>
<div class="section level2">
<h2 id="describe-annotation-table-format">Describe annotation table format:<a class="anchor" aria-label="anchor" href="#describe-annotation-table-format"></a>
</h2>
<ul>
<li>table for parallel group design</li>
<li>table for paired analysis</li>
<li>table for factorial analysis</li>
<li>additional column with explicit contrasts specification</li>
</ul>
</div>
<div class="section level2">
<h2 id="describe-yaml-file-with-application-parameters-">Describe yaml file with application parameters.<a class="anchor" aria-label="anchor" href="#describe-yaml-file-with-application-parameters-"></a>
</h2>
<ul>
<li>show example shiny application which can be used to generate yaml
file, validate inputs and run the analysis</li>
</ul>
</div>
<div class="section level2">
<h2 id="describe-generated-ouptuts">Describe generated ouptuts<a class="anchor" aria-label="anchor" href="#describe-generated-ouptuts"></a>
</h2>
<div class="section level3">
<h3 id="describe-html-document-produced">Describe html document produced:<a class="anchor" aria-label="anchor" href="#describe-html-document-produced"></a>
</h3>
<ul>
<li>Introduction:</li>
<li>B-fabric related information</li>
<li>Results
<ul>
<li>Peptide and Protein identification</li>
<li>Missing Value Analysis</li>
<li>Protein Abundance Analysis</li>
</ul>
</li>
<li>Differential Expression Analysis
<ul>
<li>Differentially Expressed Proteins</li>
</ul>
</li>
<li>Additional Analysis</li>
</ul>
</div>
<div class="section level3">
<h3 id="describe-additional-files">Describe additional files<a class="anchor" aria-label="anchor" href="#describe-additional-files"></a>
</h3>
<p>The zip file contains an excel file <strong>Results.xlsx</strong>.
All the figures can be recreated using the data in the excel file. The
Excel file contains the following spreadsheets:</p>
<ul>
<li>
<strong>annotation</strong> - the annotation of the samples in the
experiment</li>
<li>
<strong>raw_abundances</strong> table with empirical protein
abundances.</li>
<li>
<strong>normalized_abundances</strong> table with normalized protein
abundances.</li>
<li>
<strong>raw_abundances_matrix</strong> A table where each column
represents a sample and each row represents a protein and the cells
store the empirical protein abundances.</li>
<li>
<strong>normalized_abundances_matrix</strong> A table where each
column represents a sample and each row represents a protein and the
cells store the empirical protein abundances.</li>
<li>
<strong>diff_exp_analysis</strong> A table with the results of the
differential expression analysis. For each protein there is an row
containing the estimated difference between the groups, the false
discovery rate FDR, the 95% confidence interval, the posterior degrees
of freedom.</li>
<li>
<strong>missing_information</strong> - spreadsheet containing
information if a protein is present (1) or absent in a group (0).</li>
<li>
<strong>protein_variances</strong> - spreadsheet which for each
protein shows the variance (var) or standard deviation (sd) within a
group, the number of samples (n) and the number of observations (not_na)
as well as the group average intensity (mean).</li>
</ul>
<p>The zip file also includes an “boxplot.pdf” file which shows the
boxplots of normalized protein abundances for all the proteins in the
dataset.</p>
<p>The data can be used to perform functional enrichment analysis <span class="citation">[@monti2019proteomics]</span>. To compare the obtained
results with known protein interactions we recommend the <a href="https://string-db.org/" class="external-link">string-db.org</a> <span class="citation">[@Szklarczyk2017]</span>, which is a curated database
of protein-protein interaction networks for a large variety of
organisms. To simplify the data upload to string-db we include text
files containing the uniprot ids:</p>
<ul>
<li>
<code>ORA_background.txt</code> all proteins.</li>
<li>
<code>ORA_&lt;contrast_name&gt;.txt</code> proteins accepted with
the FDR and difference threshold.</li>
</ul>
<p>Other web applications allowing to run over representation analysis
(ORA) <span class="citation">[@monti2019proteomics]</span> are:</p>
<ul>
<li><a href="https://david.ncifcrf.gov/home.jsp" class="external-link">DAVID Bioinformatics
Resource</a></li>
<li>
<a href="http://www.webgestalt.org" class="external-link">WEB-based GEne SeT AnaLysis
Toolkit</a> <span class="citation">[@Wang2017]</span>
</li>
</ul>
<p>Furthermore Protein IDs sorted by t-statistic can then be subjected
to gene set enrichment analysis (GSEA) <span class="citation">[@subramanian2005gene]</span>. To simplify running GSEA
we provide the file :</p>
<ul>
<li><code>GSEA_&lt;contrast_name&gt;.rnk</code></li>
</ul>
<p>This file can be used with the webgestalt web application or used
with the GSEA application from <a href="https://www.gsea-msigdb.org/" class="external-link">gsea-msigdb</a></p>
<p>For questions and improvement suggestions, with respect to this
report, please do contact <a href="mailto:protinf@fgcz.uzh.ch" class="email">protinf@fgcz.uzh.ch</a>.</p>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Witold Wolski, Leonardo Schwarz.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.2.0.</p>
</div>

    </footer>
</div>





  </body>
</html>
